### PR TITLE
lexer: Rework godocs

### PR DESCRIPTION
**Pull request by juliaogris**
Rework package, public type and function godocs for lexer package
according to new guidelines. While writing docs it became obvious that
we could remove several bits of this package make others private - less
documentation effort and commitment to maintain compatibility.

---

To view docs locally pull branch and run

    go run golang.org/x/pkgsite/cmd/pkgsite@latest -open .

### CODE REVIEW

**Review by camh- — APPROVED** [review]
:green_circle:

File: pkg/lexer/lexer.go [hunk 1]
```diff
@@ -17,10 +32,14 @@ type Lexer struct {
 	col   int
 }
 
+// New creates a new Lexer for the given input string.
 func New(input string) *Lexer {
 	return &Lexer{input: []rune(input), pos: -1, line: 1}
 }
 
+// Next returns the next [Token] in the input string. When the end of
+// the input string is reached Next repeatedly returns a Token with
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply pkg/lexer/lexer.go:41]: ok, agreed, fixed.

File: pkg/lexer/token.go [hunk 2]
```diff
@@ -5,6 +5,13 @@ import (
 	"strconv"
 )
 
+// Token contains
+//   - type of the token, such as [IDENT], [PLUS] or [NUM_LIT]
+//   - start location in the input string
+//   - literal value of the token, if applicable
+//
+// The literal value is only relevant for the number and string literals
+// as well as comments.
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply pkg/lexer/token.go:14]: ok, updated.

File: pkg/lexer/lexer.go [hunk 3]
```diff
@@ -1,14 +1,29 @@
...
+// responsible for this task. It provides a [Lexer] type, which can be
+// initialized using the [New] function. The [Lexer.Next] method
+// returns the next Token in the input string. A [Token] is a data
+// structure that represents a single token in the input code. The
+// [EOF] token is a special token that indicates the end of the input
+// code.
+//
+// The [parser] then takes these tokens and creates an Abstract Syntax
+// Tree(AST), which is a representation of the Evy code's structure.
+// Finally, the [evaluator] walks the AST and executes the program.
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply pkg/lexer/lexer.go:15]: You raise a valid point, but I'd still like to keep it. There is no higher-level document (at the moment) that would capture this - arguably I could stick it in the evy command main package but it probably belongs there even less.
To me lexer, parser and evaluator closely work together and really need to be understood in the context of each other. I'd really like to for the evy source code to be as approachable as possible too, so I'd like to leave it in, but I'm happy to improve the wording if you can tell me how/what you think is not good?

[the remark this answers is not in the document]
Comment (juliaogris) [reply pkg/lexer/lexer.go:15]: I intend to leave a similar 3 sentence overview with all three packages (lexer, parser, evaluator).

[the remark this answers is not in the document]
Comment (camh-) [reply pkg/lexer/lexer.go:15]: Ok. no worries

**Review by camh- — APPROVED** [review]
:green_circle:

**Pull request merged**